frameworks/bitcoin-move/sources/ord.move

```diff
@@ -286,8 +292,19 @@ module bitcoin_move::ord {
         let store_obj_id = object::named_object_id<InscriptionStore>();
         let store_obj = object::borrow_mut_object_shared<InscriptionStore>(store_obj_id);
         // record a sequence_number to InscriptionID mapping
-        object::add_field(store_obj, inscription.sequence_number, id);
+        let sequence_number = inscription.sequence_number;
+        let metaprotocol = inscription.metaprotocol;
+        object::add_field(store_obj, sequence_number, id);
         let object = object::new_with_parent_and_id(store_obj, id, inscription);
+        let obj_id = object::id(&object);
+        if (option::is_some(&metaprotocol)) {
+            let metaprotocol = option::destroy_some(metaprotocol);
+            moveos_std::event_queue::emit(metaprotocol, NewInscriptionEvent{
```

Comment:
For the same metaprotocol, are other types of events allowed to be sent, such as inscription transfer events and inscription destruction events? These events need to be written to the same queue in order.

Reply:
Define an event with action?
```
struct InscriptionEvent{
  action: "new"|"transfer",
  inscription_obj_id: ObjectID,
  sequence_number: u32
}
```

Reply:
This can meet the needs